### PR TITLE
Changes in provenance lib to support replay of result collections

### PR DESCRIPTION
**Pull request by colinvwood**

### CODE REVIEW

**Comment by colinvwood** [comment]
blocked by #726

File: qiime2/core/archive/provenance_lib/tests/test_replay.py [hunk 1]
```diff
@@ -923,6 +903,243 @@ def test_build_import_usage_python(self):
...
+        cls.pm = PluginManager()
+        cls.dp = cls.pm.plugins['dummy-plugin']
+
+        cls.single_int = Artifact.import_data('SingleInt', 0)
+        cls.dict_of_ints = cls.dp.methods['dict_of_ints']
+        cls.list_of_ints = cls.dp.methods['list_of_ints']
+
+    @classmethod
+    def tearDownClass(cls):
+        pass
```
Comment (colinvwood) [inline qiime2/core/archive/provenance_lib/tests/test_replay.py]: remove this

File: qiime2/core/archive/provenance_lib/tests/test_replay.py [hunk 2]
```diff
@@ -923,6 +903,239 @@ def test_build_import_usage_python(self):
...
+            dag = ProvDAG(in_fp)
+
+            out_fp = pathlib.Path(tempdir) / 'rendered.txt'
+            out_fn = str(out_fp)
+            replay_provenance(ReplayPythonUsage, dag, out_fn)
+
+            with open(out_fp) as fh:
+                rendered = fh.read()
+
+        exp = '''\
```
Comment (fabianegli) [inline qiime2/core/archive/provenance_lib/tests/test_replay.py:1098]: Check out textwrap.dedent. It might help to format this string nicer.

**Review by ebolyen — APPROVED** [review]
This was a real doozy. Nice work. I discussed a few things we can refactor with @colinvwood, but we can merge this as is to get some testing with it.

**Pull request merged**